src/core/componentSystems/StatsSystem.js

```diff
@@ -0,0 +1,59 @@
+import { mrjsUtils } from 'mrjs';
+import { MRSystem } from 'mrjs/core/MRSystem';
+import { MRStats } from 'mrjs/core/entities/MRStats';
+
+import Stats from 'stats.js';
+
+import { HTMLMesh } from 'three/addons/interactive/HTMLMesh.js';
+import { InteractiveGroup } from 'three/addons/interactive/InteractiveGroup.js';
+
+// TODO: JSDoc
+
+export class StatsSystem extends MRSystem {
+    constructor() {
+        super(false);
+        this.statsEntities = [];
+    }
+
+    update() {
+        for (const entity of this.statsEntities) {
+            if (entity.stats === null && this.app) {
+                entity.stats = new Stats();
+                entity.stats.showPanel(0); // 0: fps, 1: ms, 2: mb, 3+: custom
+                document.body.appendChild(entity.stats.dom);
+
+                // Note: InteractiveGroup API is refactored in the newer Three.js rev.
+                //       Don't forget to update the followings when upgrading Three.js
+                entity.interactiveGroup = new InteractiveGroup(this.app.renderer, this.app.camera);
+                entity.object3D.add(entity.interactiveGroup);
+
+                // Note: HTMLMesh seems to have a bug that image is not displayed at a correct position.
+                //       in Three.js r161 or older.
+                // TODO: Upgrade Three.js to fix
+
+                // Question: Circle pointer doesn't seem to be displayed on the Stats panel object
+                //           and the clickable area seems to be a difference place from the panel object, why?

```

Comment:
@michaelthatsit @hanbollar Do you have any clue for this problem?

Reply:
not entirely sure at first glance - i'll take a quick look in headset and get back to you on that

Reply:
my first thought would be - it might be due to the following not interacting with each other nicely: [document item creation](https://github.com/Volumetrics-io/mrjs/pull/508/files#diff-9a9b14ca4cde264f45d74bc2e9e2926a929c81d6fccf4161133d5fc99d48d3faR117-R122) and [document child append](https://github.com/Volumetrics-io/mrjs/pull/508/files#diff-2dec46b40149b872e74789e7c24f5881c4b2443f3246f06414bbf697819e55b2R23)

where you have both the html item and the threejs version of it showing up (with only one being recognized as the clickable item - and with the adjustment for position from [here](https://github.com/Volumetrics-io/mrjs/pull/508/files#diff-9a9b14ca4cde264f45d74bc2e9e2926a929c81d6fccf4161133d5fc99d48d3faR120) )

----

but i'm not 100% sure it makes full sense that that will cause the issue you're experiencing with the rest of what you currently have implemented

Reply:
the other thing is - since this pr is not entirely following the ECS setup yet in the system, it might be hitting a weird case where on creation, some of the entity/dom items arent getting interpreted in the overall setup properly

once the `mr-stats` and `MRSystem` are in a better ECS setup, it'll probably be clearer as to why this issue is happening